module/apmgrpc/client.go

```diff
@@ -83,6 +83,10 @@ func NewUnaryClientInterceptor(o ...ClientOption) grpc.UnaryClientInterceptor {
 					Resource: url.Host,
 				})
 			}
+			span.Context.SetServiceTarget(apm.ServiceTargetSpanContext{
+				Type: "grpc",
```

Comment:
```suggestion
```

Reply:
Same issue, test is failing because it's not an exit span. Should we update this in the issue to omit`service.target` for non-exit spans?

Reply:
> Same issue, test is failing because it's not an exit span. Should we update this in the issue to omitservice.target for non-exit spans?

Ah, and we don't have a way to mark some gRPC spans as exit spans yet...

Yes, let's just update this PR to omit for non-exit spans.

Reply:
The span context doesn't know if the parent span is an exit span. I'm not sure how to proceed, should we add a field to the span context ? I'd prefer to avoid that as it feels like coupling different components

Reply:
> should we add a field to the span context ? I'd prefer to avoid that as it feels like coupling different components

I also don't think we add the exit flag as a field. Here's a couple of options that come to mind:

1. Update `Span.End` to do something like this:

```go
if s.exit {
        // The span was created as an exit span, but the user did not
        // manually set the service.target fields.
        s.setExitSpanServiceTarget()
} else {
        s.clearExitSpanServiceTarget()
}
```

Maybe like this:

```go
func (s *Span) End() {
    ...
    s.setExitSpanServiceTarget()
    ...
}

func (s *Span) setExitSpanServiceTarget() {
    if !s.exit {
        // Clear any manually set `service.target.*` context.
        s.Context.SetServiceTarget(ServiceTargetSpanContext{})
        return
    }

    fallbackType := s.Subtype 
    if fallbackType == "" {
        fallbackType = s.Type
    }

    // Service target fields explicitly provided.
    if s.Context.setServiceTargetCalled {
    ...
}

func (sc *SpanContext) SetServiceTarget(target ServiceTargetSpanContext) {
    c.setServiceTargetCalled = true
    if service.Name == "" {
        s.model.Service = nil
        return
    }
    c.serviceTarget.Type = truncateString(service.Type)
    c.serviceTarget.Name = truncateString(service.Name)
    c.service.Target = &c.serviceTarget
    c.model.Service = &c.service
}
```


2. We could pass `span.exit` into `SpanContext.build()` in `buildModelSpan`:

https://github.com/elastic/apm-agent-go/blob/99546a3cffd13007614d18aa926bf4dcc6c8d240/modelwriter.go#L156

And `build()` would clear the `model.SpanContext.Service` field for non-exit spans.

Reply:
Thank you for help! 🙇 

Quick question before pushing more changes: do we have a way to mark grpc/http requests as exit spans ? The latest change is going to reduce coverage of this feature because the spans used in testing are not exit spans.

Reply:
> Quick question before pushing more changes: do we have a way to mark grpc/http requests as exit spans ? The latest change is going to reduce coverage of this feature because the spans used in testing are not exit spans.

Not at the moment. For gRPC, we always assume that the target service is instrumented and pass trace context. Thus it should not be considered an "exit" span. I think we should make this configurable. I opened an issue about doing this for `apmhttp` spans recently, and have just updated it to also take `apmgrpc` client spans into account: https://github.com/elastic/apm-agent-go/issues/1304

Reply:
No worries 😄 

I think this PR grew in scope quite a lot, I've spli the fix in a separate PR: https://github.com/elastic/apm-agent-go/pull/1317